### DISCUSSION TITLE
稚晖大大，改一下.gitattributes就能让GitHub的语言统计识别Altium文件

### DISCUSSION THREAD

**Discussion by Mehver** [peng-zhihui/ElectronBot#206] [Show and tell]
![image](https://user-images.githubusercontent.com/75297777/180629355-afefd3e4-55ec-4b92-bdbe-049e919b5df1.png)

效果类似上图，就是能让GitHub的语言统计识别到这些EDA软件。我整理了一个万能的`.gitattributes`，可以使GitHub正确的识别Altium Designer、KiCAD、Gerber以及Eagle。我把这个文件放在了我的Gist里[https://gist.github.com/TitanRGB/61c37bf71159cc9a511558ec7c218339](https://gist.github.com/TitanRGB/61c37bf71159cc9a511558ec7c218339)

其中，能使AD语言生效的部分是：
```
*.OutJob linguist-detectable=true
*.PcbDoc linguist-detectable=true
*.PrjPCB linguist-detectable=true
*.SchDoc linguist-detectable=true
*.outjob linguist-detectable=true
*.pcbdoc linguist-detectable=true
*.prjpcb linguist-detectable=true
*.schdoc linguist-detectable=true
```
直接把这几行放到.gitattributes里就行。

其原理可见[https://github.com/github/linguist](https://github.com/github/linguist)，这就是GitHub语言统计功能的repo。

让GitHub统计这些EDA为语言除了满足强迫症，更重要的是很多人检索项目时，通过显示的“主要语言”能更方便的区分软件项目和硬件项目。很多时候我想找硬件设计的项目，但是看到语言是`C`或者`C++`，点进去才知道是固件之类的；如果看到语言是某个EDA，就能更快的检索到想找的项目。此外，通过“指定语言搜索”会更加方便！

**Comment by Kitem** [comment]
您好，我是焕挺，您的邮件已经收到，我会尽快回复！

**Reply by peng-zhihui** [reply]
ok学习了，多谢

**Reply by Mehver** [reply]
大大这么快就回复，太开心了！今天刚好看到新视频，没想到您也开始用立创EDA了。GitHub添加新语言的门槛是有200个repo，所以将来立创的项目多了之后，也是可以添加到语言识别中的！

另外想请教一下，您是已经完全转向立创专业版了吗？我以前没画过复杂的板子，不知道现在立创专业版的DRC和布线是否可以完全取代AD？还是说您选择立创更多的是出于免费软件的考虑？立创目前只能导出obj，还不支持stl，所以这次键盘项目的外壳设计是怎么跑通的？是先有键盘外壳尺寸才画的PCB吗，还是导出DXF画外壳？

**Reply by Mehver** [reply]
要不我直接PR提交一下这个文件？不想的话就算啦

**Reply by peng-zhihui** [reply]
1.立创EDA专业版支持step导出的
2.你提吧

**Reply by Mehver** [reply]
#208 
提交了，其他项目我也一起PR了吧。就不关联issue了

**Comment by Mehver** [comment]
@peng-zhihui 

先提交了一批。另外，要想让.gitattributes生效，需要等服务器的缓存更新。所以不会马上看到效果。

效果好的（能统计到且比例协调）：
https://github.com/peng-zhihui/MiniRover-Hardware/pull/12
https://github.com/peng-zhihui/Project-Quantum/pull/50
https://github.com/peng-zhihui/Planck-Pi/pull/18
https://github.com/peng-zhihui/ST-Link-Nano/pull/10
https://github.com/peng-zhihui/CycloidAcuratorNano/pull/5

效果还行的（能统计到Altium，但是只占10~20%，比例不协调）：
https://github.com/peng-zhihui/Dummy-Robot/pull/144
https://github.com/peng-zhihui/ElectronBot/pull/208
https://github.com/peng-zhihui/Ctrl-FOC-Lite/pull/24

效果扑街的（统计到的AD和Gerber比例不协调 / 或者比例小到被忽略不计）：
https://github.com/peng-zhihui/ESP32-PicoDK/pull/28

**Reply by Mehver** [reply]
给大佬提交PR，粉丝圆梦了哈哈哈哈哈哈
